nixos/plasma6: fix fingerprint unlocking

nbdd0121 opened the pull request:
KDE will launch two parallel PAM sessions, one expecting password and one expecting fingerprint. When fprintd is enabled, `kde.fprintd` will default to true and we're actually expecting fingerprint login from both PAM sessions, which actually block password login until pam-fprintd timeouts.

The example in https://invent.kde.org/plasma/kscreenlocker/-/merge_requests/163 shows that the "kde" pam config should not include `fingerprint-auth`.

This PR solves the unlocking part of #239770, however it won't solve the SDDM login part of it, as SDDM doesn't implement the parallel PAM sessions for logging in (while KDE screenlocker does).

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nbdd0121 committed:
nixos/plasma6: fix fingerprint unlocking

KDE will launch two parallel PAM sessions, one expecting password and
one expecting fingerprint. When fprintd is enabled, `kde.fprintd` will
default to true and we're actually expecting fingerprint login from both
PAM sessions, which actually block password login until pam-fprintd
timeouts.

K900 commented:
Ughhhh. Yeah. God I hate PAM. Also probably need to do the same for kde-smartcard then...

nbdd0121 committed:
nixos/plasma6: fix smartcard unlocking

Similar to fprint changes, ensure that kde-smartcard is dedicated to
smartcard and other two PAM configs don't use smartcard.

nbdd0121 commented:
Added the smartcard change, although I can't test it.
